### PR TITLE
fix for ShapefileReader NPE [GEOT-5262]

### PR DESCRIPTION
**Pull request by ianturton**
Fix issue with null bounding box for some filters in ShapefileReader. See https://osgeo-org.atlassian.net/projects/GEOT/issues/GEOT-5262

### CODE REVIEW

**Comment by aaime** [comment]
The test file seems to have been fully reformatted?

**Comment by ianturton** [comment]
Damm, I thought I'd trained eclipse to stop doing that. New machine sigh
I'll fix it

**Comment by ianturton** [comment]
Ignore the disappearance and reappearance of the ImageWorked - hit delete in the wrong window

File: modules/plugin/shapefile/src/test/java/org/geotools/data/shapefile/ShapefileDataStoreTest.java [hunk 1]
```diff
@@ -520,6 +520,29 @@ public void testFidFilter() throws Exception {
...
+        SimpleFeatureCollection features;// = featureSource.getFeatures();
+        
+
+        
+        GeometryFactory geometryFactory = new GeometryFactory();
+        Coordinate coordinate = new Coordinate(-99.0,38.0);
+        Point p = geometryFactory.createPoint(coordinate);
+        FilterFactory2 ff = CommonFactoryFinder.getFilterFactory2(null);
+        final Filter testFilter = ff.intersects(ff.literal(p), ff.property("the_geom"));
+        System.out.println(testFilter);
```
Comment (aaime) [inline modules/plugin/shapefile/src/test/java/org/geotools/data/shapefile/ShapefileDataStoreTest.java]: No system.out in tests please (I use the too, but try to remember to comment them out before committing)

**Comment by aaime** [comment]
Looks good minus a minor problem in the tests

**Pull request merged**